lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -739,13 +739,15 @@ class NxosCmdRef:
           multiplier: 3
     """
 
-    def __init__(self, module, cmd_ref_str):
+    def __init__(self, module, cmd_ref_str, **kwargs):
         """Initialize cmd_ref from yaml data."""
+        ref_only = kwargs.get('ref_only', False)
+
```

Comment:
Why this formulation instead of
``` python
def __init__(self, module, cmdref, ref_only=False):
```

Reply:
Originally I had something similar except that I set `ref_only` to None but one of my team members (@chrisvanheuveln ) asked for the following change:

> I'd prefer using optional keyword args, not a fan of positional options and I have a feeling this module will grow its options before long.
> e.g.
> 
> ```
>     def __init__(self, module, cmd_ref_str, **kwargs):
>         ...
>         ref_only = kwargs.get('ref_only', False)
> ```
> 
> Callers can then:
> 
> ```
>     NxosCmdRef(self._module, TMS_GLOBAL, ref_only=True)
> ```

Reply:
The way I wrote it is an optional keyword argument.  This is the way it would be written as a non-optional positional argument:
``` python
def __init__(self, module, cmdref, ref_only):
```

Using ```**kwargs```` is less readable and should only be used if there's a special reason for it (such as when a function is a light wrapper around a different function and simply passes all of those keyword arguments on to the wrapped function)

@chrisvanheuveln might have been mixing terms, though... maybe he can quickly say why he thinks ```**kwargs``` is the better choice here?